src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -110,6 +111,48 @@ static uint16_t               eq_count_max;
 static uint16_t               eq_count;
 static uint16_t               eq_idx;
 
+/* the number of EQs for aio contexts. (aio_eq_count + eq_count) <= eq_count_max */
+static uint16_t               aio_eq_count;
+
+struct aio_eq {
+	daos_handle_t    eq;
+	pthread_mutex_t  lock;
+	_Atomic uint64_t n_op_queued;
+	_Atomic uint64_t n_op_done;
+};
+
+/* list of EQs dedicated for aio contexts. */
+static struct aio_eq aio_eq_list[MAX_EQ];
+/* the accumulated iodepth for aio context. */
+static long int      depth_sum_accu;
+
+struct aio_ev {
+	daos_event_t    ev;
+	struct iocb    *piocb;
+	struct aio_ctx *ctx;
+};
+
+struct aio_ctx {
+	/* the real io_context_t used in libaio */
+	io_context_t     ctx;
+	/* the depth of context set by io_setup */
+	int              depth;
+	int              first_eq;
+	bool             inited;
+	bool             on_dfs;
+	_Atomic uint64_t num_op_submitted;
+	_Atomic uint64_t num_op_done;
+	/* DFS is involved or not for current context */
+	pthread_mutex_t  lock;
```

Comment:
Does this comment match the entry or not?  The comment implies that it's a boolean but the entry is a lock?